### PR TITLE
Initial commit for Babelfish v3.1

### PR DESCRIPTION
**Pull request by rishabhtanwar29**
### Description
* Updated Github action scripts.
* Created 15_1 test folder.
* Updated T-SQL extension version to 3.1.0 and added two upgrade scripts:
   * babelfishpg_tsql--3.0.0--3.1.0.sql
   * babelfishpg_tsql--2.4.0--3.0.0.sql (Duplicate of babelfishpg_tsql--2.3.0--3.0.0.sql)
* Disabled BABEL-SP_FKEYS test from 15 schedule files due to BABEL-3837
* Views sys.views and sys.sp_columns_100_view return incorrect/no result due to
`collate sys.default_collation` clause so replaced it with `CAST` to proper type.

Signed-off-by: Rishabh Tanwar <ritanwar@amazon.com>

### Test Scenarios Covered ###
* **Use case based -**

* **Boundary conditions -**

* **Arbitrary inputs -**

* **Negative test cases -**

* **Minor version upgrade tests -**

* **Major version upgrade tests -**

* **Performance tests -**

* **Tooling impact -**

* **Client tests -**

### Check List
- [X] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is under the terms of the Apache 2.0 and PostgreSQL licenses, and grant any person obtaining a copy of the contribution permission to relicense all or a portion of my contribution to the PostgreSQL License solely to contribute all or a portion of my contribution to the PostgreSQL open source project.

For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/babelfish-for-postgresql/babelfish_extensions/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: contrib/babelfishpg_tsql/sql/babelfishpg_tsql.sql [hunk 1]
```diff
@@ -379,7 +379,7 @@ CREATE OR REPLACE VIEW sys.sp_columns_100_view AS
      )
      , sys.translate_pg_type_to_tsql(a.atttypid) AS tsql_type_name
      , sys.spt_datatype_info_table AS t5
-  WHERE (t4."DATA_TYPE" = t5.TYPE_NAME COLLATE sys.database_default)
+  WHERE (CAST(t4."DATA_TYPE" AS varchar(20)) = t5.TYPE_NAME)
```
Comment (kuntalghosh) [inline contrib/babelfishpg_tsql/sql/babelfishpg_tsql.sql]: This should be sys.varchar, right?

**Review by kuntalghosh — CHANGES_REQUESTED** [review]

File: contrib/babelfishpg_tsql/sql/babelfishpg_tsql.sql (hunk 1 above)
Comment (Deepesh125) [reply contrib/babelfishpg_tsql/sql/babelfishpg_tsql.sql]: Yes, should be sys.varchar

Comment (rishabhtanwar29) [reply contrib/babelfishpg_tsql/sql/babelfishpg_tsql.sql]: `t5.TYPE_NAME` => `sys.spt_datatype_info_table.TYPE_NAME` is `pg_catalog.varchar` (charater varying)

Comment (Deepesh125) [reply contrib/babelfishpg_tsql/sql/babelfishpg_tsql.sql]: In that case, let's cast t5.TYPE_NAME to sys.nvarchar(128).

Comment (rishabhtanwar29) [reply contrib/babelfishpg_tsql/sql/babelfishpg_tsql.sql]: Done

**Review by Deepesh125 — APPROVED** [review]

**Review by shardgupta — APPROVED** [review]

**Pull request merged**